components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LoggingResource.java

```diff
@@ -263,19 +276,54 @@ private JSONObject getLoggerData(org.apache.axis2.context.MessageContext axis2Me
         return jsonBody;
     }
 
-    private JSONArray getAllLoggerDetails(org.apache.axis2.context.MessageContext axisMsgCtx) throws IOException {
-
-        JSONArray payload = new JSONArray();
+    private static String[] getAllLoggers() throws IOException {
+        //along with root logger
+        String[] loggers = getLoggers().split(",");
         // add root logger
-        JSONObject data = getLoggerData(axisMsgCtx, Constants.ROOT_LOGGER);
-        payload.put(data);
+        int fullLength = loggers.length + 1;
+        String[] allLoggers = new String[fullLength];
+        allLoggers[0] = ROOT_LOGGER;
+        for (int i = 1; i<fullLength; i++) {
+            allLoggers[i] = loggers[i-1];
+        }
+        return allLoggers;
+    }
+    private void getAllLoggerDetails(MessageContext messageContext) throws IOException {
+        String[] loggers = getAllLoggers();
+        setResponseBody(Arrays.asList(loggers), messageContext);
+    }
 
-        String[] loggers = getLoggers().split(",");
-        for (String logger : loggers) {
-            data = getLoggerData(axisMsgCtx, logger.trim());
-            payload.put(data);
+    private static List<String> getSearchResults(String searchKey) throws IOException {
+        String[] allLoggers = getAllLoggers();
+        List<String> filteredLoggers = new ArrayList<>();
+
+        for (String logger : allLoggers) {
+            if (logger.toLowerCase().contains(searchKey)) {
+                filteredLoggers.add(logger);
+            }
+        }
+        return filteredLoggers;
+    }
+
+    private void getAllLoggerDetails(MessageContext messageContext, String searchKey) throws IOException {
+        List<String> resultsList = getSearchResults(searchKey);
+        setResponseBody(resultsList, messageContext);
+    }
+
+
+    private void setResponseBody(List<String> logConfigsList, MessageContext messageContext) {
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+
+        JSONObject jsonBody = Utils.createJSONList(logConfigsList.size());
+
+        JSONObject data;
+        for (String logger : logConfigsList) {
+            data = getLoggerData(axis2MessageContext, logger.trim());
+            jsonBody.getJSONArray(Constants.LIST).put(data);
```

Comment:
Can this be null?

Reply:
it cannot be. In the getProperty Function it does a null check